resources/assets/js/app/components/DirectionsMap.js

```diff
@@ -38,31 +38,65 @@ export default class DirectionsMap extends React.Component {
             this.props.to.lng === nextProps.to.lng &&
             this.props.from.lat === nextProps.from.lat &&
             this.props.to.lng === nextProps.to.lng
+            && !this.isWaypointsChanged(nextProps.waypoints)
         ) {
             return;
         }
 
         this.renderDirection(nextProps);
     }
 
+    isWaypointsChanged(waypoints) {
+        let result = false;
+
+        if (!waypoints) {
+            return result;
+        }
+
+        if (waypoints.length !== this.props.waypoints.length) {
+            return true;
+        }
+
+        this.props.waypoints.forEach((waypoint, index) => {
+            if (
+                waypoint.location.lat !== waypoints[index].location.lat ||
+                waypoint.location.lng !== waypoints[index].location.lng
+            ) {
+                result = true;
+            }
+        });
+
+        return result;
+    }
+
     renderDirection(props) {
         this.DirectionsService.route({
             origin: props.from,
             destination: props.to,
+            waypoints: props.waypoints || [],
             travelMode: google.maps.TravelMode.DRIVING,
         }, (result, status) => {
             if (status === google.maps.DirectionsStatus.OK) {
-                let response = result.routes[0].legs[0];
+                let route = result.routes[0];
```

Comment:
create class that will accept on construct "result" var and will have getters for all needed properties like - getRoute, getRouteStartPoint, getRouteEndPoint, getDuration etc.